File: src/main/resources/output/IFC4X3_vbnet/Types/IfcCurveStyleFontSelect.vb
' SELECT TYPE IfcCurveStyleFontSelect
' Options:
' - IfcCurveStyleFont
' - IfcPreDefinedCurveFont
Public Class IfcCurveStyleFontSelect
    Public ReadOnly Property Value As Object

    Public Sub New(value As Object)
        If Not (TypeOf value Is IfcCurveStyleFont OrElse TypeOf value Is IfcPreDefinedCurveFont) Then
            Throw New ArgumentException("Value must be one of: IfcCurveStyleFont, IfcPreDefinedCurveFont")
        End If
        Me.Value = value
    End Sub

    Public Function GetValueAs(Of T)() As T
        Return CType(Value, T)
    End Function
End Class

File: src/main/resources/output/IFC4X3_csharp/Types/IfcCurveStyleFontSelect.cs
// SELECT TYPE IfcCurveStyleFontSelect
public class IfcCurveStyleFontSelect {
    public object Value { get; }

    public IfcCurveStyleFontSelect(object value) {
        if (!(value is IfcCurveStyleFont || value is IfcPreDefinedCurveFont))
        {
            throw new ArgumentException("Value must be one of: IfcCurveStyleFont, IfcPreDefinedCurveFont");
        }
        Value = value;
    }

    public T GetValueAs<T>() {
        return (T)Value;
    }
}
